src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -767,6 +803,7 @@ private void Sweep()
             }
 
             gcDebt = totalAllocated > GC_THRESHOLD ? totalAllocated : GC_THRESHOLD;
+            sweepSet.Clear();
```

Comment:
I didn't see this change mentioned in your PR?

Reply:
the `SweepSet` is used to capture the state of the heap when a GC cycle starts up. It basically holds the heap indexes that are to be considered for cleanup by the current GC cycle.
It used to be overridden only when another GC cycle started up (with no reason). Since I am using it in the new string allocation logic I need to clean it at the end of every GC cycle.

Reply:
do you need to also get rid of the reassignment call - or thats fine?

Reply:
I refactored the code to not do reassignment all the time